### PR TITLE
Use the correct environment variable in integration tests

### PR DESCRIPTION
**Pull request by aeisenberg**
Also, fix documentation and launch configs to specify the correct and
complete set of environment variables we should be using.

This should fix the failing integration tests on our internal repository.

Replace this with a description of the changes your pull request makes.

## Checklist

- [n/a] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [n/a Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-codeql` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -77,18 +77,26 @@ $ vscode/scripts/code-cli.sh --install-extension dist/vscode-codeql-*.vsix # if
 
 You can use VS Code to debug the extension without explicitly installing it. Just open this directory as a workspace in VS Code, and hit `F5` to start a debugging session.
 
-### Running the unit/integration tests
+### Running the unit tests and integration tests that do not require a CLI instance
 
-Ensure the `CODEQL_PATH` environment variable is set to point to the `codeql` cli executable.
-
-Outside of vscode, run:
+Unit tests and many integration tests do not Outside of vscode, run:
```
Comment (adityasharad) [inline CONTRIBUTING.md:82]:
```suggestion
Unit tests and many integration tests do not require a copy of the CodeQL CLI. Outside of vscode, run:
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -122,13 +122,12 @@ jobs:
         working-directory: extensions/ql-vscode
         if: matrix.os == 'ubuntu-latest'
         run: |
-          CODEQL_PATH=$GITHUB_WORKSPACE/codeql-home/codeql/codeql npm run test
```
Comment (adityasharad) [inline .github/workflows/main.yml:125]: So these tests are not using the CLI? Should we remove it from being installed in the step above?

Comment (aeisenberg) [reply .github/workflows/main.yml:125]: They may have at one point in the past, but not any more. I will remove the section.

**Review by adityasharad — APPROVED** [review]

**Comment by adityasharad** [comment]
Updated the branch protection rules for `main`: we now require `Test (ubuntu-latest)` and `Test (windows-latest)` rather than the `{stable,nightly}` variants of each.

**Pull request merged**